api/product.rb

```diff
@@ -91,6 +91,24 @@ def set_properties_based_on_version(version)
     end
     # rubocop:enable Naming/AccessorMethodName
 
+    def exists_at_version_or_lower(name)
+      return true if @versions.nil?
+      name ||= 'ga'
+      return false unless Version::ORDER.include?(name)
+      (0..Version::ORDER.index(name)).each do |i|
+        return true if exists_at_version(Version::ORDER[i])
+      end
+      false
+    end
+
+    def exists_at_version(name)
+      return true if @versions.nil?
+      @versions.each do |v|
```

Comment:
I think you could use `return @versions.any?{|v| v.name == name}`

Reply:
Done